File: frontend/src/backend/compraService.js
import {config, appFetch} from './appFetch';

export const buy = (sesionId, quantity, creditCard, onSuccess, 
    onErrors) => {
    
        appFetch(`/compras/buy`, 
        config('POST', {sesionId, quantity, creditCard}), onSuccess, onErrors);

    }

export const getCompras = ({userId, page}, onSuccess) =>
    appFetch(`/compras?userId=${userId}&page=${page}`, config('GET'), onSuccess);

export const deliverTicket = (compraId, creditCard, onSuccess,
    onErrors) => 
    appFetch(`/compras/${compraId}/entregarTicket`,
    config('POST', {creditCard}), onSuccess, onErrors);

File: backend/src/main/java/es/udc/paproject/backend/model/services/CompraService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.exceptions.PeliculaEmpezadaException;
import es.udc.paproject.backend.model.exceptions.SalaLlenaException;
import es.udc.paproject.backend.model.exceptions.TarjetaInvalidaException;
import es.udc.paproject.backend.model.exceptions.TarjetaNoCorrespondeException;
import es.udc.paproject.backend.model.exceptions.TicketEntregadoException;
import es.udc.paproject.backend.model.entities.Compra;

public interface CompraService {

	Compra buy(Long userId, Long sesionId, int quantity, String creditCard)
			throws InstanceNotFoundException, SalaLlenaException, PeliculaEmpezadaException, TarjetaInvalidaException;

	Block<Compra> findComprasByUser(Long userId, int page, int size);

	void entregarEntradas(Long compraId, String tarjeta) throws InstanceNotFoundException, PeliculaEmpezadaException,
			TicketEntregadoException, TarjetaNoCorrespondeException;
}
